src/Version.h

```diff
@@ -17,26 +18,80 @@
 
 namespace Version {
 
-using exact_width_mumble_raw_version_t = std::uint32_t;
-using mumble_raw_version_t             = unsigned int;
+enum class CompareMode { AtLeast, LessThan };
 
-static_assert(sizeof(exact_width_mumble_raw_version_t) <= sizeof(mumble_raw_version_t),
-			  "Invalid scalar type used to represent Mumble versions");
+//
+// The mumble version format (v2) is a uint64 and consists of the following:
+// reserved/unused  major   minor   patch
+// 0xFFFF           0xFFFF  0xFFFF  0xFFFF
+// (big-endian)
+//
+
+using mumble_raw_version_t             = std::uint64_t;
+
+constexpr std::uint64_t OFFSET_UNUSED = 48;
+constexpr std::uint64_t OFFSET_MAJOR  = 32;
+constexpr std::uint64_t OFFSET_MINOR  = 16;
+constexpr std::uint64_t OFFSET_PATCH  = 0;
+constexpr std::uint64_t FIELD_WIDTH   = 0xFFFF;
```

Comment:
```suggestion
constexpr std::uint64_t FIELD_WIDTH   = std::numeric_limits< std::uint16_t >::max();
```
Also I think that `FIELD_WIDTH` is a bit misleading here. As the _width_, I would expect 2 bytes or 16 bit. But what this variable actually holds, is a field's maximum value. So maybe `FIELD_MAX_VALUE` or something like this would be a better name? :thinking:

Reply:
While initially it simply represented the bit field width of any given component, it is obviously also the max value, yes. Maybe I can come up with a better name.

Reply:
``FIELD_MASK`` or ``FIELD_BITMASK``?. I would rather not use uint16 max here, even if it is technically the same. The explicit ``0xFFFF`` makes it easier to understand the shifts in my opinion.

Reply:
`FIELD_MASK` sounds good.

Hm... I'm generally not very fond of hexadecimal numbers, but that might just be me not being used to using them.
But imo a binary representation might be more obvious than the hexadecimal one?

Reply:
Binary? You mean ``0b1111111111111111`` instead of ``0xFFFF``? No, I think everyone who worked with bit masks before will understand ``0xFFFF`` much better. But i will rename it ``FIELD_MASK`` then. That is a more accurate description.

Reply:
> You mean 0b1111111111111111 instead of 0xFFFF?

Yes

> No, I think everyone who worked with bit masks before will understand 0xFFFF much better

Alright, then :+1: